sky/global_user_state.py

```diff
@@ -408,3 +549,33 @@ def get_storage() -> List[Dict[str, Any]]:
             'status': StorageStatus[status],
         })
     return records
+
+
+def get_cost_for_cluster(cluster_name: str,) -> float:
+
+    cluster_hash = get_cluster_hash(cluster_name) or str(uuid.uuid4())
```

Comment:
We should return `None` when the `cluster_name` does not exist. Also, what does it mean when querying usage_intervals with a newly generated cluster hash in L557?

Reply:
This edit somehow was placed in the wrong spot. I meant to do "get_cluster_hash(cluster_name) or str(uuid.uuid4())" in the add_or_update_clusters method as you originally suggested, but I think I accidentally placed it here instead.